sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/descriptor/MetricDescriptor.java

```diff
@@ -94,36 +94,35 @@ public String getAggregationName() {
     return AggregationUtil.aggregationName(getView().getAggregation());
   }
 
-  @Memoized
+  /** Uses case-insensitive version of {@link #getName()}. */
   @Override
-  public abstract int hashCode();
+  public final int hashCode() {
+    // TODO: memoize
+    int h = 1;
+    h *= 1000003;
+    h ^= getName().toLowerCase(Locale.ROOT).hashCode();
+    h *= 1000003;
+    h ^= getDescription().hashCode();
+    h *= 1000003;
+    h ^= getView().hashCode();
+    h *= 1000003;
+    h ^= getSourceInstrument().hashCode();
+    return h;
+  }
 
-  /**
-   * Returns true if another metric descriptor is compatible with this one.
-   *
-   * <p>A metric descriptor is compatible with another if the following are true:
-   *
-   * <ul>
-   *   <li>{@link #getName()} is equal
-   *   <li>{@link #getDescription()} is equal
-   *   <li>{@link #getAggregationName()} is equal
-   *   <li>{@link InstrumentDescriptor#getName()} is equal
-   *   <li>{@link InstrumentDescriptor#getDescription()} is equal
-   *   <li>{@link InstrumentDescriptor#getUnit()} is equal
-   *   <li>{@link InstrumentDescriptor#getType()} is equal
-   *   <li>{@link InstrumentDescriptor#getValueType()} is equal
-   * </ul>
-   */
-  public boolean isCompatibleWith(MetricDescriptor other) {
-    return getName().equals(other.getName())
-        && getDescription().equals(other.getDescription())
-        && getAggregationName().equals(other.getAggregationName())
-        && getSourceInstrument().getName().equals(other.getSourceInstrument().getName())
-        && getSourceInstrument()
-            .getDescription()
-            .equals(other.getSourceInstrument().getDescription())
-        && getSourceInstrument().getUnit().equals(other.getSourceInstrument().getUnit())
-        && getSourceInstrument().getType().equals(other.getSourceInstrument().getType())
-        && getSourceInstrument().getValueType().equals(other.getSourceInstrument().getValueType());
+  /** Uses case-insensitive version of {@link #getName()}. */
+  @Override
+  public final boolean equals(Object o) {
+    if (o == this) {
+      return true;
+    }
+    if (o instanceof MetricDescriptor) {
+      MetricDescriptor that = (MetricDescriptor) o;
+      return this.getName().equalsIgnoreCase(that.getName())
+          && this.getDescription().equals(that.getDescription())
+          && this.getView().equals(that.getView())
+          && this.getSourceInstrument().equals(that.getSourceInstrument());
+    }
```

Comment:
The `aggregationName` used in the previous implementation is part of the view, so that's why it's not needed here. Just remarking because it gave me a small pause.